confidential/secrets_manager.py

```diff
@@ -57,7 +57,7 @@ def decrypt_secret_from_aws(self, secret_name) -> str:
                 raise Exception("We can't find the resource that you asked for.") from e
 
         else:
-            if "SecretString" not in get_secret_value_response:
+            if "SecretString" not in get_secret_value_response or not get_secret_value_response["SecretString"]:
```

Comment:
By the way, `0`, `""` (empty string) and `None` all return falsy. So let's explicitly check for `None`:

```suggestion
            if "SecretString" not in get_secret_value_response or get_secret_value_response["SecretString"] is None:
```